### PR TITLE
Allow jobs to use lightweight checkout

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Extract most of the code from Pipeline-Build-Test-Any-Platform.groovy
  and Build-Any-Platform.groovy into new file Pipeline-Initialize.groovy.
- Move remaining functionality into pipeline-functions.groovy,
  variables-functions.groovy and build.groovy
- This allows the jobs to be configured to use lightweight
  checkout will will reduce the load on master.
- The side effect is the remaining code in Pipeline-Initialize.groovy
  will not be testable via PR builds.
- The remainder of the pipeline code can still be pulled from
  any passed repo/branch/ref. Which will allow PR builds
  to continue testing pipeline changes including defaults.yml.

[skip ci]
Related #8252
Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by pshipton** [comment]
Script approval request https://bugs.eclipse.org/bugs/show_bug.cgi?id=553778

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
jenkins compile alinux64xl jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile alinux64xl jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile alinux64xl jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile alinux64xl jdk11

**Comment by AdamBrousseau** [comment]
jenkins compile alinux64xl jdk11

**Comment by AdamBrousseau** [comment]
@pshipton I don't want this to go in as-is. There's a few issues to fix that I've worked around to get it functioning at the moment. We can stress test it after the nightly finishes with a large PR build.

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11,jdk14

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11,jdk14

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11,jdk14

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11

**Comment by AdamBrousseau** [comment]
Jenkins compile all jdk8,jdk11

**Comment by pshipton** [comment]
For the record, the stress testing went well, as it didn't cause jenkins to restart. Running 4 "compile all jdk8,jdk11", a couple or more concurrently.

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
Refactored this a bit. Waiting for script approval to test it.

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
@vsebe for review.
Note there is a TEMP commit on top of the changes.

File: buildenv/jenkins/common/variables-functions.groovy [hunk 1]
```diff
@@ -1261,6 +1283,8 @@ def create_job(JOB_NAME, SDK_VERSION, SPEC, downstreamJobType, id){
     params.put('VENDOR_CREDENTIALS_ID_DEFAULT', VENDOR_CREDENTIALS_ID_DEFAULT)
     params.put('jobType', downstreamJobType)
     params.put('DISCARDER_NUM_BUILDS', DISCARDER_NUM_BUILDS)
+    params.put('SCM_REPO', 'https://github.com/Adambrousseau/openj9.git')
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions.groovy]: Hard-coded personal fork, please remove (set in Job template)

File: buildenv/jenkins/common/variables-functions.groovy [hunk 2]
```diff
@@ -771,6 +771,27 @@ def add_pr_to_description() {
     }
 }
 
+def setup() {
+    set_job_variables(params.JOB_TYPE)
+
+    switch (params.JOB_TYPE) {
+        case "pipeline":
+            buildFile = load 'buildenv/jenkins/common/pipeline-functions.groovy'
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions.groovy]: `pipeline-functions.groovy` is loaded twice: here as `buildFile` and at L203:` pipelineFunctions = load 'buildenv/jenkins/common/pipeline-functions.groovy'`
Could use: `buildFile = pipelineFunctions `

File: buildenv/jenkins/jobs/pipelines/Pipeline-Initialize.groovy [hunk 3]
```diff
@@ -57,18 +57,12 @@ timeout(time: 10, unit: 'HOURS') {
                         userRemoteConfigs: [remoteConfigParameters]]
 
                 variableFile = load 'buildenv/jenkins/common/variables-functions.groovy'
```
Comment (vsebe) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Initialize.groovy]: This script loads `pipeline-functions.groovy` multiple times:
variables-functions.groovy loads pipeline-functions.groovy (twice: see L203, L779)
build.groovy also loads pipeline-functions.groovy.
Some reformatting might be required in order to load the file only once. Could be done in a different PR.

Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Initialize.groovy]: Opened an issue for myself to revisit this later.

File: buildenv/jenkins/common/variables-functions.groovy (hunk 2 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy]: Changed

File: buildenv/jenkins/common/variables-functions.groovy (hunk 1 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy]: As per my previous comment. Will remove once testing is done.

**Comment by AdamBrousseau** [comment]
Jenkins compile zlinux jdk8

**Comment by AdamBrousseau** [comment]
Seems to still work. Any other concerns @vsebe ?

**Comment by AdamBrousseau** [comment]
Ok I will remove the extra commit and WIP label

**Pull request merged**